vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_list_reader.cc

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/publisher/publisher_list_reader.h"
+
+#include <utility>
+
+#include "bat/ledger/internal/publisher/prefix_util.h"
+
+namespace braveledger_publisher {
+
+PublisherListReader::PublisherListReader()
+    : prefix_size_(kMinPrefixSize) {}
+
+PublisherListReader::~PublisherListReader() = default;
+
+PublisherListReader::ParseError PublisherListReader::Parse(
+    const std::string& contents) {
+  using publishers_pb::PublisherList;
+
+  PublisherList message;
+  if (!message.ParseFromString(contents)) {
+    return ParseError::InvalidProtobufMessage;
+  }
+
+  size_t prefix_size = message.prefix_size();
+  if (prefix_size < kMinPrefixSize || prefix_size > kMaxPrefixSize) {
+    return ParseError::InvalidPrefixSize;
+  }
+
+  size_t uncompressed_size = message.uncompressed_size();
+  if (uncompressed_size == 0) {
+    return ParseError::InvalidUncompressedSize;
+  }
+
+  std::string uncompressed;
+  switch (message.compression_type()) {
+    case PublisherList::NO_COMPRESSION: {
+      uncompressed = std::move(*message.mutable_prefixes());
+      break;
+    }
+    default: {
```

Comment:
can we remove this default? So that if we add new enum compiler will return an error and we will not miss it during implementation

Reply:
Unfortunately, no. Removing the switch results in a compiler error:

```
error: enumeration values 'PublisherList_CompressionType_PublisherList_CompressionType_INT_MIN_SENTINEL_DO_NOT_USE_' and 'PublisherList_CompressionType_PublisherList_CompressionType_INT_MAX_SENTINEL_DO_NOT_USE_' not handled in switch
```

Also, after parsing the field can also contain invalid values outside of the range of the declared enum that we need to handle. I'll add a test case for this.